File: Amazon_LeetCode_Python_C++/ArrayString/238_product_of_array_except_self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1 for i in range(len(nums))]
        mul = 1
        
        for i, num in enumerate(nums):
            ans[i] *= mul
            mul *= num
            
        mul = 1    
        for i in range(len(nums) - 1, -1, -1):
            ans[i] *= mul
            mul *= nums[i]
            
        return ans

File: Amazon_LeetCode_Python_C++/ArrayString/238_product_of_array_except_self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size(), 1);
        int mul = 1;
        
        for (int i = 0; i < nums.size(); i++) {
            ans.at(i) *= mul;
            mul *= nums.at(i);
        }
        
        mul = 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            ans.at(i) *= mul;
            mul *= nums.at(i);
        }
        
        return ans;
    }
};